test/test-utils/gameManager.ts

```diff
@@ -473,8 +473,8 @@ export class GameManager {
    */
   switchPokemon(pokemonIndex: number): void {
     this.onNextPrompt("CommandPhase", UiMode.COMMAND, () => {
-      (this.scene.ui.getHandler() as CommandUiHandler).setCursor(2);
-      (this.scene.ui.getHandler() as CommandUiHandler).processInput(Button.ACTION);
+      (this.scene.ui.getCurrentHandler() as CommandUiHandler).setCursor(2);
+      (this.scene.ui.getCurrentHandler() as CommandUiHandler).processInput(Button.ACTION);
```

Comment:
NIT: Use generic param `<T>`

```suggestion
      this.scene.ui.getCurrentHandler<CommandUiHandler>().setCursor(2);
      this.scene.ui.getCurrentHandler<CommandUiHandler>().processInput(Button.ACTION);
```